webknossos/webknossos/dataset/view.py

```diff
@@ -368,10 +369,19 @@ def some_work(args: Tuple[View, int], some_parameter: int) -> None:
 
         # execute the work for each chunk
         if executor is None:
-            for args in job_args:
-                work_on_chunk(args)
+            if progress_desc is None:
+                for args in job_args:
+                    work_on_chunk(args)
+            else:
+                with get_rich_progress() as progress:
+                    task = progress.add_task(progress_desc, total=len(job_args))
+                    for args in job_args:
+                        work_on_chunk(args)
+                        progress.update(task, advance=1)
```

Comment:
In this case we might want to weight the jobs by their bounding-box size, since partial chunks at the border of the dataset would be finished faster. I guess this might be easy to add, what do you think?

Reply:
Ideally, you are right that the progress bar should be weighted by the size of the bounding boxes. However, if we do this here, this would be a bit inconsistent with the code in `wait_and_ensure_success` (where we do not know the size of the bounding boxes). Do you think I should change this anyways?

Reply:
No strong opinion here. I think that the inconsistency is not that important, but rather the more precise logging, but also fine as-is.